UT4MasterServer/Controllers/SessionController.cs

```diff
@@ -122,10 +119,34 @@ public class SessionController : JsonAPIController
 					return ErrorInvalidRequest("password");
 				}
 
-				// TODO: Check permission and return ErrorResponse: Sorry your client is not allowed to use the grant type password. errorCode: errors.com.epicgames.common.oauth.unauthorized_client
+				// Yeah, EPIC is returning this ErrorResponse after checking username and password...
+				if (!allowPasswordGrant)
+				{
+					return BadRequest(new ErrorResponse
```

Comment:
I understand that epic is checking credentials first, but I don't think this is desired. Even though its just null checks before it, I would prefer if this was returned prior to any checks at location where `if (!allowPasswordGrant)` was before.

Maybe make a comment saying we are doing things differently than epic.

Reply:
Done